Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -168,7 +169,16 @@ class PaintedContentsInfo {
 #if HAVE(HDR_SUPPORT)
     bool paintsHDRContent()
     {
-        determinePaintsContent();
+        if (m_hdrContent == RequestState::Unknown)
+            determinePaintsContent();
+        return m_hdrContent == RequestState::True || m_hdrContent == RequestState::Undetermined;
+    }
+
+    void determinePaintsHDRContent();
+    bool paintHDRContentOnly()
```

Comment:
The naming of `determinePaintsHDRContent` needs to match the `isReplacedElementWithHDR` naming so that the "only" here is clear. I would avoid "only", because an HDR image can have a non-HDR CSS border.

Reply:
This two functions were removed. I added calculating  whether `PaintedContentsInfo` is `isReplacedElementWithHDR` to `PaintedContentsInfo::determineContentsType()`